### DISCUSSION TITLE
Plan for separating libbpfgo from tracee repo

### DISCUSSION THREAD

**Discussion by grantseltzer** [aquasecurity/tracee#727] [Announcements]
I'd like to separate libbpfgo from this repo. Since it's already its own go module we would better be able to serve projects that want to import it by having it logically separated. 

This entails

- Creating a new repo with the libbpfgo code, including the github actions workflows
- Removing libbpfgo code from this repo,  including the github actions workflows
- Updating anywhere libbpfgo is imported within tracee to the new module name
- Making sure other project consumers of libbpfgo are aware of this change

Points that I want to be very careful of:
- We should cut a new releases as soon as we move to the new repo
- I don't think we will have any problems with people using previous releases of tracee/libbpfgo even if we remove it off the tip of the main branch.

**Comment by grantseltzer** [comment]
@eyakubovich, @brancz  As contributors, and presumably consumers of libbpfgo, does this sound good to you?

**Reply by eyakubovich** [reply]
Sounds good overall. One thing to figure out is which project will "own" (either vendor or install) libbpf. libbpfgo obviously needs it to build but it can be the responsibility of the user to make sure it's installed and a path provided via env vars. Alternatively libbpfgo can vendor it itself.

**Reply by itaysk** [reply]
I think libbpfgo should just declare libbpf as a dependency, and users should decide how they provide it (vendoring, dynamic linking, etc). We should have examples in libbpfgo of the different approaches. 
Tracee, as one of the users, will keep vendoring libbpf as it has been so far.

Does this make sense to you?

**Reply by eyakubovich** [reply]
That sounds good but I want to make sure `go get` works as smooth as possible. Can we add `-lbpf` to `#cgo LDFLAGS` in the source file? If the user has `libbpf` installed in standard location, it'll just work. Otherwise they'll need to to set `CGO_LDFALGS=-L/path/to/lib` prior to `go get`.

**Reply by itaysk** [reply]
good idea, sounds like it can be a separate issue

**Comment by itaysk** [comment]
I'd just move the 2nd step to be last (removing). good point about removing from head shouldn't affect existing users who use the commit hash.

**Comment by grantseltzer** [comment]
Alright #728 will be the last commit into libbpfgo before separating it into its own repo. I will cut a release as soon as I make that repo public.

As for tracee, I will update usage to the new repo, then subsequently remove libbpfgo from it.

**Reply by grantseltzer** [reply]
Also I will transfer all issues

**Comment by grantseltzer** [comment]
Get it while it's hot! https://github.com/aquasecurity/libbpfgo

**Reply by brancz** [reply]
Just converted my usage to this, and works beautifully! Once it matures a bit more it might make sense to even move it to the iovisor or libbpf org. In any case, I'll continue on those contributions now! :)
